### PR TITLE
Add Concrete TES unit model

### PR DESCRIPTION
**Pull request by radhakrishnatg**
## Addresses issue:
N/A

## Summary/Motivation:
#68 is being replaced by this PR.

This PR adds the unit model for concrete thermal energy storage (TES), and a model for supercritical pulverized coal-fired power plant integrated with concrete TES. Concrete TES system is modeled as shown in the figure below:

image

image

## Changes proposed in this PR:
- Added a unit model for concrete thermal energy storage system
- Added a flowsheet model for supercritical power plant integrated with concrete TES.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md and COPYRIGHT.md file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Comment by radhakrishnatg** [comment]
@nareshsusarla @esrawli @jghouse88 Could you please review this PR when you have some time? Thank you!

**Comment by ksbeattie** [comment]
@nareshsusarla will review this PR this week.

File: dispatches/unit_models/concrete_tes.py [hunk 1]
```diff
@@ -0,0 +1,964 @@
...
+        # Add the heat transfer coefficient surrogate model
+        def htc_surrogate():
+            # TODO: Need to declare a variable for outer radius of the
+            #       concrete block and do the following calculation in a constraint
+            # Also, do this inside each hex model instead of doing it
+            # outside. Reduces a lot of additional constraints!
+            face_area = data["face_area"]
+            a = data["tube_diameter"] / 2  # Inner radius of the concrete block
+            b = sqrt(face_area / Constants.pi + a ** 2)  # Outer radius
+            k_red = 0.8  # Reduction factor for thermal conductivity
```
Comment (nareshsusarla) [inline dispatches/unit_models/concrete_tes.py:712]: Add a reference for the value, if there is one, else if it is just an assumption then you can state that.

File: dispatches/unit_models/concrete_tes.py [hunk 2]
```diff
@@ -0,0 +1,964 @@
...
+            k = data["therm_cond_concrete"] * k_red
+            correction_factor = 1.31
```
Comment (nareshsusarla) [inline dispatches/unit_models/concrete_tes.py:714]: Same as above for k_red

**Review by nareshsusarla — APPROVED** [review]
Only minor comments rest looks good

**Comment by adowling2** [comment]
Ready to merge.

**Pull request merged**